charts/actions-runner-controller/templates/githubwebhook.service.yaml

```
@@ -6,6 +6,10 @@ metadata:
   namespace: {{ .Release.Namespace }}
   labels:
     {{- include "actions-runner-controller.labels" . | nindent 4 }}
+{{- if .Values.githubWebhookServer.service.annotations }}
+  annotations:
+{{ toYaml .Values.githubWebhookServer.service.annotations | nindent 4 }}
```

Hey! Thanks for the PR. I thought this had to have 4 spaces if you use `nindent 4`?

```suggestion
    {{ toYaml .Values.githubWebhookServer.service.annotations | nindent 4 }}
```

Otherwise, `annotations: {"a":"a","b":"b"}` might have wrong indentation like:

```
  annotations:
a: a
    b: b
```

---

Hey @mumoshu, thanks for replying so quickly (and I appreciate the support on a seemingly small issue).

I ran some tests and it looks like the templating of go removes the indentations before adding the `nindent` regardless of the original indentation level. I went searching a bit and it looks like other oss projects do the same (i.e k8s dashboard here: https://github.com/kubernetes/dashboard/blob/master/aio/deploy/helm-chart/kubernetes-dashboard/templates/deployment.yaml#L19-L22). Let me know what you think, I'd be happy to make the edits.

---

Generally you'd start at the beginning of the line with `indent`, `nindent` allows you add the indentation after a new line (so the current indent is _reset_ so to speak) and it looks a lot more yaml-y to the trained eye. I *think* the empty line would get parsed as valid yaml and fixed and helm templating time.

---

@rlandesman @sledigabel Wow thanks a lot! I've been misunderstanding how it works. 

https://github.com/kubernetes/dashboard/blob/master/aio/deploy/helm-chart/kubernetes-dashboard/templates/deployment.yaml#L19-L22 uses `indent 4` instead of `nindent 4` which is used here, but I learned that both works.

I'd still appreciate it if you could have 4 spaces like:

```
  annotations:
    {{ toYaml .Values.githubWebhookServer.service.annotations | nindent 4 }}
```

so that it is consistent with other parts of the chart.

I once thought it might be even better to write it like:

```
  annotations: {{ toYaml .Values.githubWebhookServer.service.annotations | nindent 4 }}
```

But this turns out not to compose well when you want to add more fields to be considered:

```
  annotations: {{ toYaml .Values.githubWebhookServer.service.annotations | nindent 4 }}
    {{ toYaml $otherCommonAnnotations | nindent 4 }}
```

---

Sure thing, changed it 👍🏼